shared/js/background/trackers.es6.js

```diff
@@ -220,7 +195,20 @@ function checkTrackersWithParentCompany (url, siteDomain, request) {
     })
 
     if (toBlock) {
-        return toBlock
+        // we have a blocked tracker but need to see if it should be whitelisted
+        if (toBlock.trackerData && toBlock.trackerData.whitelist) {
+            // check whitelist
+            toBlock.trackerData.whitelist.some(ruleObj => {
+                if (requestMatchesRule(request, ruleObj.rule) && matchRuleOptions(ruleObj, request, siteDomain)) {
+                    toBlock = null
+                }
+            })
```

Comment:
this looks suspicious 🤔 we are modifying `toBlock` inside of the loop which is going over data from `toBlock`?

BTW this will go through the whole list every time (even if whitelisted entry is first). we can avoid that by doing:

```js
const foundOnWhitelist = toBlock.trackerData.whitelist.some(ruleObj => {
    if (requestMatchesRule(request, ruleObj.rule) && matchRuleOptions(ruleObj, request, siteDomain)) {
        return true;
    }
    return false;
})
```

Reply:
Me too. I spent some time reorganizing the whole algorithm today. Still needs some more work but it's looking better.